cmd/clidoc/main.go

```diff
@@ -192,25 +194,82 @@ func writeMessages(path string) error {
 }
 
 func validateAllMessages(path string) error {
+	type message struct {
+		ID, Name string
+	}
+
+	usedIDs := make([]message, 0, len(messages))
 	set := token.NewFileSet()
 	packs, err := parser.ParseDir(set, path, nil, 0)
 	if err != nil {
 		return errors.Wrapf(err, "unable to parse text directory")
 	}
+	info := &types.Info{
+		Defs: make(map[*ast.Ident]types.Object),
+	}
+	var pack *ast.Package
+	for _, p := range packs {
+		if p.Name == "text" {
+			pack = p
+			break
+		}
+	}
+	allFiles := make([]*ast.File, 0)
+	for fn, f := range pack.Files {
+		if strings.HasSuffix(fn, "/id.go") {
+			allFiles = append(allFiles, f)
+		}
+	}
+	_, err = (&types.Config{Importer: importer.Default()}).Check("text", set, allFiles, info)
```

Comment:
This gave me a bit of a hard time, as I could not get it to properly resolve imports from modules. Therefore, the warning in the `id.go` file.